lib/pages/search_page.dart

```diff
@@ -1,50 +1,308 @@
 import 'package:flutter/material.dart';
+import 'package:go_ride_sharing/pages/map_page.dart';
 import 'package:go_ride_sharing/services/post_service.dart';
+import 'package:go_ride_sharing/theme.dart';
+import 'package:go_ride_sharing/widgets/map_window.dart';
 import 'package:go_ride_sharing/widgets/post_card.dart';
-import 'package:go_ride_sharing/models/post.dart'; // Assuming you have a Post model
-
+import 'package:go_ride_sharing/models/post.dart';
+import 'package:google_maps_flutter/google_maps_flutter.dart';
+import 'package:go_ride_sharing/models/search_filter.dart';
 class SearchPage extends StatefulWidget {
+  const SearchPage({super.key});
+
   @override
-  _SearchPageState createState() => _SearchPageState();
+  SearchPageState createState() => SearchPageState();
 }
 
-class _SearchPageState extends State<SearchPage> {
+class SearchPageState extends State<SearchPage> {
   Future<List<Post>>? _postsFuture;
+  final _seatsAvailableController = TextEditingController();
+  final _departureDateController = TextEditingController();
+  final _priceController = TextEditingController();
+
+  LatLng? origin;
+  LatLng? destination;
+  Map<MarkerId, Marker> markers = {};
 
   @override
   void initState() {
     super.initState();
     _postsFuture = PostService().fetchAllPosts();
   }
 
+  @override
+  void dispose() {
+    // Dispose controllers to free up resources
+    _seatsAvailableController.dispose();
+    _departureDateController.dispose();
+    _priceController.dispose();
+    super.dispose();
+  }
+
+  // Function to select a date using a date picker
+  Future<void> _selectDate(BuildContext context) async {
+    final DateTime? picked = await showDatePicker(
+      context: context,
+      initialDate: DateTime.now(),
+      firstDate: DateTime(2000),
+      lastDate: DateTime(2101),
+    );
+    if (picked != null) {
+      setState(() {
+        _departureDateController.text = "${picked.toLocal()}".split(' ')[0];
+      });
+      print("Date selected: ${_departureDateController.text}");
+    }
+  }
+
+  Future<void> _navigateAndDisplayMap(BuildContext context) async {
+    // Navigator.push returns a Future that completes after calling
+    // Navigator.pop on the Selection Screen.
+    Map<MarkerId, Marker> result = await Navigator.push(
+      context,
+      MaterialPageRoute(builder: (context) => MapPage()),
+    );
+
+    // When a BuildContext is used from a StatefulWidget, the mounted property
+    // must be checked after an asynchronous gap.
+    if (!context.mounted) return;
+
+    setState(() {
+      markers = result;
+      origin = result.values.first.position;
+      destination = result.values.last.position;
+    });
+  }
+  // void _applyFilters() {
```

Comment:
I'm seeing a lot of comments like this one as well as many TODO comments everywhere.

Is this feature 100% done?

Reply:
Good catch. Yeah, the feature is functional, I will go back and clean up the comments.